cfme/configure/tasks.py

```diff
@@ -70,39 +167,27 @@ def is_task_finished(destination, task_name, expected_status, clear_tasks_after_
     return True
 
 
-def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
-    """ Check if analysis is finished - if not, reload page"""
+def wait_analysis_finished_multiple_tasks(
+        task_name, task_type, expected_num_of_tasks, delay=5, timeout='5M'):
+    """ Wait until analysis is finished (or timeout exceeded)"""
+    row_completed = []
+    # get view for reload button
+    view = navigate_to(Tasks, 'AllTasks')
 
-    tabs_data = {
-        'container': {
-            'tab': 'AllTasks',
-            'task': '{}',
-            'state': 'finished'
-        },
-        'vm': {
-            'tab': 'AllTasks',
-            'task': 'Scan from Vm {}',
-            'state': 'finished'
-        },
-        'host': {
-            'tab': 'MyOtherTasks',
-            'task': "SmartState Analysis for '{}'",
-            'state': 'Finished'
-        },
-        'datastore': {
-            'tab': 'MyOtherTasks',
-            'task': 'SmartState Analysis for \[{}\]',
-            'state': "Finished"
-        },
-        'cluster': {
-            'tab': 'MyOtherTasks',
-            'task': 'SmartState Analysis for \[{}\]',
-            'state': "Finished"}
-    }[task_type]
-    return is_task_finished(destination=tabs_data['tab'],
-                            task_name=tabs_data['task'].format(name),
-                            expected_status=tabs_data['state'],
-                            clear_tasks_after_success=clear_tasks_after_success)
+    def tasks_finished(output_rows=row_completed, task_name=task_name,
```

Comment:
**Required**

Secondary, this is a poor pattern, defining this local method with all of its defaults set to local variables. Either use the `@wait_for` decorator instead of the call, or actually pass func_args/func_kwargs list/dictionary to `wait_for()`

Reply:
Remove all default values from method.
Added func_kwargs to wait_for()